lib/rake/build.rake

```diff
@@ -111,17 +111,13 @@ namespace :build do
         # The sequencing described here is the best for mitigating any issues
         # that may arise when that best practice is not followed.
         ChatClient.log 'Restarting <b>dashboard</b> Active Job worker(s).'
+        # Issue a stop command to all workers. Will kill if not stopped within ~20 seconds.
+        RakeUtils.system 'bin/delayed_job', 'stop'
```

Comment:
you will need the `-n` here:
```
Dave-MBP:~/src/cdo2 (_staging2)$ ps -ef | grep delayed_job | grep -v grep
Dave-MBP:~/src/cdo2 (_staging2)$ ./dashboard/bin/delayed_job -n 2 start
GetSecretValue: development/cdo/slack_bot_token
delayed_job.0: process with pid 62354 started.
delayed_job.1: process with pid 62358 started.
Dave-MBP:~/src/cdo2 (_staging2)$ ps -ef | grep delayed_job | grep -v grep
  501 62354     1   0  3:45PM ??         0:03.44 delayed_job.0    
  501 62358     1   0  3:45PM ??         0:00.44 delayed_job.1    
Dave-MBP:~/src/cdo2 (_staging2)$ ./dashboard/bin/delayed_job stop        
GetSecretValue: development/cdo/slack_bot_token
Dave-MBP:~/src/cdo2 (_staging2)$ ps -ef | grep delayed_job | grep -v grep
  501 62354     1   0  3:45PM ??         0:26.47 delayed_job.0    
  501 62358     1   0  3:45PM ??         0:16.25 delayed_job.1    
Dave-MBP:~/src/cdo2 (_staging2)$ ./dashboard/bin/delayed_job -n 2 stop   
GetSecretValue: development/cdo/slack_bot_token
delayed_job.0: trying to stop process with pid 62354 sending TERM and waiting 20s ...
delayed_job.0: process with pid 62354 successfully stopped.
delayed_job.1: trying to stop process with pid 62358 sending TERM and waiting 20s ...
delayed_job.1: process with pid 62358 successfully stopped.
Dave-MBP:~/src/cdo2 (_staging2)$ ps -ef | grep delayed_job | grep -v grep
Dave-MBP:~/src/cdo2 (_staging2)$ 
```

Reply:
Just kidding... I forgot to upgrade the gem first! looks good!
```
Dave-MBP:~/src/cdo2 (darin/activejob-stop-start)$ ./dashboard/bin/delayed_job -n 2 start  
GetSecretValue: development/cdo/slack_bot_token
Dave-MBP:~/src/cdo2 (darin/activejob-stop-start)$ ./dashboard/bin/delayed_job stop      
GetSecretValue: development/cdo/slack_bot_token
delayed_job: trying to stop process with pid 63207...
delayed_job: trying to stop process with pid 63205...
delayed_job: process with pid 63205 successfully stopped.
delayed_job: process with pid 63207 successfully stopped.
Dave-MBP:~/src/cdo2 (darin/activejob-stop-start)$ ps -ef | grep delayed_job | grep -v grep
```

Reply:
for testing, I am wondering if we need something in place so that we do not silently regress in case someone decides to upgrade this gem? maybe a test which does basically what I've just done manually?

Reply:
Added a comment to the line in the Gemfile.